2005-01-21  Michael Banck  <mbanck@debian.org>

        * Makefile ($(libdir)/libpthread2.a): Install linker script
        from $(srcdir) instead of current directory.
        ($(libdir)/libpthread2_pic.a): Likewise.

Index: Makefile
===================================================================
RCS file: /cvsroot/hurd/hurd/libpthread/Makefile,v
retrieving revision 1.2
diff -u -r1.2 Makefile
--- libpthread/Makefile	20 Aug 2003 20:40:06 -0000	1.2
+++ libpthread/Makefile	20 Jan 2005 23:35:31 -0000
@@ -181,14 +181,16 @@
 .PHONY: $(libdir)/libpthread.a $(libdir)/libpthread_pic.a
 
 # XXX: These rules are a hack.  But it is better than messing with
-# ../Makeconf at the moment.
+# ../Makeconf at the moment.  Note that the linker scripts
+# $(srcdir)/libpthread.a and $(srcdir)/libpthread_pic.a get overwritten
+# when building in $(srcdir) and not a seperate build directory.
 $(libdir)/libpthread2.a: $(libdir)/libpthread.a
 	mv $< $@
-	$(INSTALL_DATA) libpthread.a $<
+	$(INSTALL_DATA) $(srcdir)/libpthread.a $<
 
 $(libdir)/libpthread2_pic.a: $(libdir)/libpthread_pic.a
 	mv $< $@
-	$(INSTALL_DATA) libpthread_pic.a $<
+	$(INSTALL_DATA) $(srcdir)/libpthread_pic.a $<
 
 .PHONY: $(addprefix $(includedir)/, $(sysdeps_headers))
 
